public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -280,16 +283,69 @@ export class LokiDatasource
 
     if (fixedRequest.liveStreaming) {
       return this.runLiveQueryThroughBackend(fixedRequest);
-    } else {
-      const startTime = new Date();
-      return super.query(fixedRequest).pipe(
-        // in case of an empty query, this is somehow run twice. `share()` is no workaround here as the observable is generated from `of()`.
-        map((response) =>
-          transformBackendResult(response, fixedRequest.targets, this.instanceSettings.jsonData.derivedFields ?? [])
-        ),
-        tap((response) => trackQuery(response, fixedRequest, startTime))
-      );
     }
+
+    if (requestSupportsPartitioning(fixedRequest.targets)) {
+      return this.partitionedQuery(fixedRequest);
+    }
+
+    const startTime = new Date();
+    return super.query(fixedRequest).pipe(
+      // in case of an empty query, this is somehow run twice. `share()` is no workaround here as the observable is generated from `of()`.
```

Comment:
Fyi, we talked with @svennergr in the past how to use `share` with hot observables https://raintank-corp.slack.com/archives/C03PMJLVC67/p1670335801702649